internal/cli/auth/register.go

```diff
@@ -15,20 +15,76 @@
 package auth
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"os"
 
 	"github.com/mongodb/mongocli/internal/cli/require"
 	"github.com/mongodb/mongocli/internal/config"
+	"github.com/pkg/browser"
 	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
+const accountURI = "https://account.mongodb.com/account/register?fromURI=https://account.mongodb.com/account/connect"
+const govAccountURI = "https://account.mongodbgov.com/account/register?fromURI=https://account.mongodbgov.com/account/connect"
+
 type RegisterOpts struct {
 	loginOpts
 }
 
-func (opts *RegisterOpts) Run() error {
-	_, _ = fmt.Fprintf(opts.OutWriter, "Create and verify your MongoDB Atlas account from the web browser and return to Atlas CLI after activating your account.\n")
-	// TODO: CLOUDP-120669
+func (opts *RegisterOpts) registerAndAuthenticate(ctx context.Context) error {
+	// TODO:CLOUDP-121210 - Replace with new request and remove URI override.
+	code, _, err := opts.flow.RequestCode(ctx)
+	if err != nil {
+		return err
+	}
+
+	if opts.isGov {
+		code.VerificationURI = govAccountURI
+	} else {
+		code.VerificationURI = accountURI
+	}
+
+	opts.printAuthInstructions(code)
+
+	if !opts.noBrowser {
+		if errBrowser := browser.OpenURL(code.VerificationURI); errBrowser != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "There was an issue opening your browser\n")
+		}
+	}
+
+	accessToken, _, err := opts.flow.PollToken(ctx, code)
+	var target *atlas.ErrorResponse
+	if errors.As(err, &target) && target.ErrorCode == authExpiredError {
+		return errors.New("authentication timed out")
```

Comment:
make the error a [sentinel error](https://go.dev/blog/go1.13-errors) and use it for both commands 
`var errTimedOut = errors.New("authentication timed out")`
```suggestion
		return errTimedOut
```